### ISSUE TITLE
Associate all objects with a casa_org_id

### ISSUE THREAD

**Issue by Zrrrpy** [rubyforgood/casa#597]
Part of epic #530

**Description**
This work should be broken into two parts:

- [ ] Identify the places where objects are not associated with `casa_org_id`
- [ ] Associate all things with `casa_org_id` that currently are not. 

Example: every time a domain object is created or called with a query, every single one of those needs to be inspected to see if it is associated with a `casa_org_id`. i.e. `volunteer.create` `volunteer.where`.

**Comment by abachman** [comment]
tl;dr - I don't think we need to add additional casa_org_id fields to anything in the system. We're in a good spot for multi-tenancy w.r.t. who owns what. 

 Right now there are two models associated with CasaOrgs: **User** (which includes Admin, Supervisor, and Volunteer) and **CasaCase**.

Of the the other three models in the system, **CaseContact**, **CaseAssignment**, and **SupervisorVolunteer**, two (CA and SV) are join tables between org-owned models and the third (CC) is only ever viewed in the context of a User or a Case. That means that once CasaOrg is part of the request cycle (e.g., specified by the `current_user` or as part of a subdomain or URL scheme), then all record look-ups will/can/should start with a User or Case and be scoped from there.

For User and Case listing routes, which is only `DashboardController#show`,  CasaOrg should be the parent class from which all users and cases are listed. 

Only CaseContact has listing routes (`DashboardController#show` and `CaseContactReportsController#index`) and those should be scoped to Cases belonging to the current CasaOrg, getting rid of the CaseContact-based lookups. 

Bottom line, adding casa_org_id to the other three models mentioned would be redundant given their existing associations and require extra code be added to every record creation statement. It can be added later if it proves necessary for a new reason, but the way records are used now, we can protect access by scoping all lookups to User or CasaCase.
